support/supportedversion/version.go

```diff
@@ -163,3 +220,113 @@ func LookupLatestSupportedRelease(ctx context.Context, hc *hyperv1.HostedCluster
+	if !ok {
+		return "<unknown>"
+	}
+
+	for _, setting := range bi.Settings {
+		if setting.Key == "vcs.revision" {
+			return setting.Value
+		}
+	}
+	return "<unknown>"
+}
+
+func String() string {
+	return fmt.Sprintf("openshift/hypershift: %s. Latest supported OCP: %s", GetRevision(), LatestSupportedVersion)
+}
+
+type SupportedVersions struct {
+	Versions []string `json:"versions"`
+}
+
+// GetSupportedOCPVersions retrieves the supported OCP versions from the server. It fetches the ConfigMap containing the
+// supported versions and the server version from the specified namespace and unmarshals the versions into a
+// SupportedVersions struct. If the ConfigMap or the required keys are not found, it returns an error. The function
+// returns the supported versions, the server version, and any error encountered during the process.
+func GetSupportedOCPVersions(ctx context.Context, namespace string, client crclient.Client) (SupportedVersions, string, error) {
+	var versions SupportedVersions
+	var serverVersion string
+
+	// Fetch the supported versions ConfigMap from the specified namespace
+	supportedVersions := manifests.ConfigMap(namespace)
+	if err := client.Get(ctx, crclient.ObjectKeyFromObject(supportedVersions), supportedVersions); err != nil {
+		return SupportedVersions{}, "", fmt.Errorf("failed to find supported versions on the server: %v", err)
+	}
+
+	// Check if the ConfigMap contains the server version key
+	if serverVersionValue, present := supportedVersions.Data[config.ConfigMapServerVersionKey]; present {
+		serverVersion = serverVersionValue
+	} else {
+		return SupportedVersions{}, "", fmt.Errorf("the server did not advertise its HyperShift version")
+	}
+
+	// Check if the ConfigMap contains the supported versions key
+	if supportedVersionData, present := supportedVersions.Data[config.ConfigMapVersionsKey]; present {
+		if err := json.Unmarshal([]byte(supportedVersionData), &versions); err != nil {
+			return SupportedVersions{}, "", fmt.Errorf("failed to parse supported versions on the server: %v", err)
+		}
+
+		return versions, serverVersion, nil
+	} else {
+		return SupportedVersions{}, "", fmt.Errorf("the server did not advertise supported OCP versions")
+	}
+}
+
+type ocpTags struct {
+	Name string       `json:"name"`
+	Tags []ocpVersion `json:"tags"`
+}
+
+// retrieveSupportedOCPVersion retrieves the latest supported OCP version from supported versions ConfigMap, retrieves
+// the latest stable release images from the provided release URL, and returns the latest supported OCP version that is
+// not a release candidate and matches the latest supported OCP version supported by the HyperShift operator.
+func retrieveSupportedOCPVersion(releaseURL string, client crclient.Client) (ocpVersion, error) {
+	var stableOCPVersions ocpTags
+
+	// Get the latest supported OCP version from the supported versions ConfigMap
+	supportedOCPVersions, _, err := GetSupportedOCPVersions(context.TODO(), "hypershift", client)
```

Comment:
I'm not sure how to get around not hardcoding this to `hypershift` for the support versions ConfigMap 🤔 

This function is called by `LookupDefaultOCPVersion`, which is called in several places where we would not know the namespace where the HyperShift Operator is installed.